app/src/main/java/com/poupa/vinylmusicplayer/provider/MusicPlaybackQueueStore.java

```diff
@@ -201,8 +202,11 @@ private synchronized void saveQueue(final String tableName, @NonNull final Array
 
     @NonNull
     private ArrayList<Song> getQueue(@NonNull final String tableName) {
-        Cursor cursor = getReadableDatabase().query(tableName, null,
-                null, null, null, null, null);
-        return SongLoader.getSongs(cursor);
+        try (Cursor cursor = getReadableDatabase().query(tableName, new String[]{BaseColumns._ID},
+                null, null, null, null, null)) {
```

Comment:
Eventhough we still save the song's metadata in the MediaStore playlist DB (see todo note above - for backward compat), only song ID is extracted back.